backend-core/src/main/java/eu/socialedge/hermes/backend/core/GoogleMapsShapeFactory.java

```diff
@@ -0,0 +1,59 @@
+package eu.socialedge.hermes.backend.core;
+
+import com.google.maps.DistanceMatrixApi;
+import com.google.maps.GeoApiContext;
+import com.google.maps.model.DistanceMatrixElementStatus;
+import com.google.maps.model.LatLng;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import tec.uom.se.quantity.Quantities;
+import tec.uom.se.unit.Units;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.commons.lang3.Validate.notEmpty;
+
+@Component
+public class GoogleMapsShapeFactory implements ShapeFactory {
```

Comment:
I believe that components in -core module must be decoupled from platform/framework implementation and from -application especially. Leave -code module without spring dependencies.